src/data/starredSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from "@reduxjs/toolkit"
+
+const starredSlice = createSlice({
+    name: 'starred',
+    initialState: {
+        starredMovies: []
+    },
+    reducers: {
+        starMovie: (state, action) => {
+            state.starredMovies = [action.payload, ...state.starredMovies]
+        },
+        unstarMovie: (state, action) => {
+            const indexOfId = state.starredMovies.findIndex(key => key.id === action.payload.id)
```

Comment:
You're receiving the whole movie as a payload but you're just using the `id` I recommend you update the logic to receive only the `id`